### ISSUE TITLE
No distinction between double and float CVs possible

### ISSUE THREAD

**Issue by jpfeuffer** [OpenMS/OpenMS#4648]
This is very bad, since other libraries depend on the precision with which numbers are written
if specified specifically by a schema.
https://github.com/jpfeuffer/OpenMS/blob/380516006d6f7bd08a797768e4c92b0da4dbb165/src/openms/include/OpenMS/FORMAT/ControlledVocabulary.h#L67-L67

**Comment by timosachsenberg** [comment]
Is this really a problem?
As I understood it, the precision of the value is independent from its lexical representation.
So even if the precision is low, the third-party library should still be able to parse a lexical representation with additional non-significant digits.

**Comment by jpfeuffer** [comment]
If it's a good library, yes.

**Comment by poshul** [comment]
My proposal is that we limit the length of `XSD_DECIMAL` output to be compliant with float (to handle library parsing issues), and circle back around to this for 3.0 with a proper fix. A little bit of digging shows that since `XS_FLOAT` and `XS_DOUBLE` are not derived from `XS_DECIMAL` making changes to how we output XML would break compatibility with current and past versions. I do believe that making this change eventually is the correct thing to do since at least MzML does differentiate between the two lengths.

**Comment by timosachsenberg** [comment]
Maybe we could add the other types to ControlledVocabulary.h, query that information in our XML handlers and write them out accordingly.
BTW can you link to the max number of digits that may be written out?

**Comment by timosachsenberg** [comment]
I would vote to make this a minor issue as it only caused a problem once in one specific deprecated java tool so far.
Or did I miss anything of importance?

**Comment by jpfeuffer** [comment]
No, it was just marked for this release. Can be minor due to low practical importance.
